src/WatchDogs.FakeSource/DataGeneratorRepeaterBackroundTask.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace WatchDogs.FakeSource;
+
+public class DataGeneratorRepeaterBackroundTask : IDataGeneratorRepeaterBackroundTask
```

Comment:
You are on the right track here, but we need to consider one thing: 
> In the future, there will be several modules for different monitoring activities and different data sources, so the application architecture should take this into account and provide corresponding level of abstractions.

If we want this, we need an interface what does basically this but it is not bound to any specific source. E. g., fake source, DxTrade, or some other source would all implement this interface but its implementation would vary based on the source (here we use Bogus, in DxTrade we would use WebSockets). I would consider simply renaming `IDataGeneratorRepeaterBackroundTask` to something like `IWatcher` or `IWatchDog` and _not_ have it as a background service.

We would still have a background service, though, which launches all the watchers. Simply another layer.

Reply:
Done.